runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java

```diff
@@ -160,11 +159,23 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
                     append(stateMachine.getName()).
                     append(".started").toString());
         } catch (ConstraintViolationException ex) {
-            //in case of Duplicate correlation key, return http code 409 conflict
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(ex.getCause() != null ? ex.getCause().getMessage() : null).build();
+            //In case of Duplicate correlation key, return http code 409 conflict
+            if(ex.getCause() != null) {
+                if (ex.getCause().getMessage().contains("Duplicate entry")) {
```

Comment:
ex.getClause().getMessage().toLowerCase().contains("duplicate entry") 

make it case agnostic.

Reply:
can you instead do this in the underlying DAO layer and throw relevant exception from their? DATA layer details are leaking to Resource layer which is exactly what DAO was supposed to abstract out.

Reply:
> ex.getClause().getMessage().toLowerCase().contains("duplicate entry")
> 
> make it case agnostic.

Done, thanks for pointing out.